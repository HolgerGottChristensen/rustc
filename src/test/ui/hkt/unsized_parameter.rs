// HKT check that the value given to the I is sized,
// check-fail

fn test<I<%J>>(_input: I<[u32]>) { } //~ ERROR the size for values of type `[u32]` cannot be known at compilation time [E0277]

fn main() {}
