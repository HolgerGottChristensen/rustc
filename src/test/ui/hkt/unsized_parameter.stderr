error[E0277]: the size for values of type `[u32]` cannot be known at compilation time
  --> $DIR/unsized_parameter.rs:4:24
   |
LL | fn test<I<%J>>(_input: I<[u32]>) { }
   |                        ^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `[u32]`
note: required by a bound in `test::I`
  --> $DIR/unsized_parameter.rs:4:12
   |
LL | fn test<I<%J>>(_input: I<[u32]>) { }
   |            ^ required by this bound in `test::I`
help: consider relaxing the implicit `Sized` restriction
   |
LL | fn test<I<%J: ?Sized>>(_input: I<[u32]>) { }
   |             ++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
